sample-apps/delivery-customizations/extensions/delivery-customization-js/src/run.js

```diff
@@ -2,20 +2,20 @@
 
 /**
  * @typedef {import("../generated/api").RunInput} RunInput
- * @typedef {import("../generated/api").FunctionRunResult} FunctionRunResult
+ * @typedef {import("../generated/api").CartDeliveryOptionsTransformRunResult} CartDeliveryOptionsTransformRunResult
  * @typedef {import("../generated/api").Operation} Operation
  */
 
 /**
- * @type {FunctionRunResult}
+ * @type {CartDeliveryOptionsTransformRunResult}
  */
 const NO_CHANGES = {
   operations: [],
 };
 
 /**
  * @param {RunInput} input
- * @returns {FunctionRunResult}
+ * @returns {CartDeliveryOptionsTransformRunResult}
```

Comment:
You're missing the change to `rename:` operation below which should be `deliveryOptionRename` now. Possibly elsewhere too.

Reply:
Oh it's the tests that also missing, which explains why it passed. I really don't know why anyone ever thought dynamic types were a good idea